src/app/home/home.component.ts

```diff
@@ -111,7 +141,8 @@ export class HomeComponent implements OnInit {
   filter() {
     if (this.userFilter != "") {
       try {
-        this.filteredProgress = this.currentProgress.filter(prog => prog.username.match(this.userFilter))
+        this.filteredProgress = this.currentProgress.filter(prog =>  prog.username.toLowerCase().match(this.userFilter.toLowerCase())) 
```

Comment:
```suggestion
        const pattern = new RegExp(this.userFilter, 'i');
        this.filteredProgress = this.currentProgress.filter(prog =>  prog.username.match(pattern)) 
```
I think the above would make it a little easier to understand what's happening.